umh-core/pkg/fsm/benthos/reconcile.go

```diff
@@ -89,7 +89,7 @@ func (b *BenthosInstance) Reconcile(ctx context.Context, snapshot fsm.SystemSnap
 	}
 
 	// Step 2: Detect external changes.
-	if err := b.reconcileExternalChanges(ctx, services, snapshot.Tick, start); err != nil {
+	if err := b.reconcileExternalChanges(ctx, services, snapshot, snapshot.Tick, start); err != nil {
```

Comment:
Since this pr already refactors the signature of many functions, we could also just pass snapshot, instead of snapshot + snapshot.Tick